ignite/contrib/handlers/ema_handler.py

```diff
@@ -0,0 +1,204 @@
+          2. Build ``trainer`` (``ignite.engine.Engine``).
+          3. Resume from checkpoint for ``model`` and ``ema_handler``.
+          4. Attach ``ema_handler`` to ``trainer``.
+          5. Attach other handlers to ``trainer``.
+
+    Note:
+          When the trainer completes running, the EMA weights are moved to the model. If both the handler's
+          ``state_dict`` and model's ``state_dict`` are saved by ``Checkpoint`` handler, users can retrieve the EMA
+          weights as follows:
+
+          .. code-block:: python
+
+              to_load = {"model" model, "ema_handler", ema_handler}
+              ckpt = torch.load("checkpoint.pt")
+              Checkpoint.load_objects(to_load, ckpt)
+
+              # EMA weights are actually in the model's state_dict
+              smoothed_weights = model.state_dict()
+
+          Consider resuming training from a saved checkpoint: although the EMA weights are in the model when loaded
+          from a checkpoint, there is no need to manually swap the ``sate_dict`` of handler and model. The Handler
+          will automatically perform this step when the training resumes.
+
+    Note:
+          Users need to register the EMA handler before the validation handler, so that swapping weights will be
+          performed before validation begins. It is recommended to register the EMA handler as the first handler for
+          the event ``Events.EPOCH_COMPLETED``.
+
+    Examples:
+          .. code-block:: python
+
+              device = torch.device("cuda:0")
+              model = nn.Linear(2, 1).to(device)
+              # update the ema every 5 iterations
+              ema_handler = EMAHandler(
+                model, momentum=0.0002, momentum_warmup=0.0001, interval=5, warmup_iters=10000, device=device)
+              trainer = Engine(step_fn)
+              to_load = {"model": model, "ema_handler", ema_handler, "trainer", trainer}
+              if resume_from is not None:
+                  Checkpoint.load_objects(to_load, checkpoint=resume_from)
+
+              # add other handlers
+              to_save = to_load
+              ckpt_handler = Checkpoint(to_save, DiskSaver(...), ...)
+              trainer.add_event_handler(Events.EPOCH_COMPLETED, ckpt_handler)
+
+              trainer.run(...)
+
+
+    """
+
+    def __init__(
+        self,
+        model: nn.Module,
+        momentum: float = 0.0002,
+        momentum_warmup: float = 0.0001,
+        interval: int = 1,
+        warmup_iters: int = 100,
+        device: Optional[Union[str, torch.device]] = None,
+    ) -> None:
+        assert 0 < momentum < 1
+        assert 0 < momentum_warmup < 1
+        assert momentum_warmup <= momentum
+        assert isinstance(interval, int) and interval > 0
+        assert isinstance(warmup_iters, int) and warmup_iters > 0
+        assert isinstance(model, nn.Module)
```

Comment:
In our codebase we use `if-not-then-raise` instead of `assert`. Could you please change that.

Reply:
Thanks. It is fixed in next commit